### ISSUE TITLE
Figures with links cause build errors

### ISSUE THREAD

**Issue by james-d-elliott** [h-enk/doks#820]
## Description

Using a figure with a link causes a build error. in 0.5.0. The issue appears to be here: https://github.com/h-enk/hyas-images/blob/master/layouts/partials/helpers/lib-output/image-handling/thumbnail-or-full-picture.html#L94

This seems to be because the `helpers/link-open` partial doesn't exist.

### Steps to reproduce

Use the following example in 0.5.0: `{{< figure link="https://www.istockphoto.com/" src="https://media.istockphoto.com/photos/young-woman-watches-sunrise-outside-camping-tent-picture-id1248575497?s=612x612" height="50" >}}`

### Expected result

Should create an image with a link.

### Actual result

<details>
  <summary>Error Output</summary>
hugo v0.101.0-466fa43c16709b4483689930a4f9ac8add5c9f66+extended windows/amd64 BuildDate=2022-06-16T07:09:16Z VendorInfo=gohugoio
Error: Error building site: "content\en\docs\prologue\commands.md:26:1": failed to render shortcode "figure": failed to process shortcode: "node_modules\@hyas\images\layouts\shortcodes\figure.html:1:4": execute of template failed: template: shortcodes/figure.ht
ml:1:4: executing "shortcodes/figure.html" at <partial "helpers/wrapped-image" (dict "width" (.Get "width") "height" (.Get "height") "alt" (.Get "alt") "title" (.Get "title") "image" (partial "helpers/lib/image-handling/find-image-src" (dict "src" (.Get "src") "page" .Page "getRelative" (.Get "getRelative" | default true) "ignoreBundleAssets" (.Get "ignor
eBundleAssets" | default false) "ignoreSiteAssets" (.Get "ignoreSiteAssets" | default false) "ignoreStaticImages" (.Get "ignoreStaticImages" | default true))) "page" .Page "link" (.Get "link") "target" (.Get "target") "rel" (.Get "rel") "imageWrapper" (default "figure" (.Get "imageWrapper")) "caption" (.Get "caption") "attr" (.Get "attr") "attrLink" (.Get
 "attrLink") "class" (.Get "class") "singleSize" (.Get "singleSize") "thumbnails" (.Get "thumbnails"))>: error calling partial: "node_modules\@hyas\images\layouts\partials\helpers\wrapped-image.html:109:8": execute of template failed: template: partials/helpers/wrapped-image.html:109:8: executing "partials/h
elpers/wrapped-image.html" at <partial "helpers/lib-output/image-handling/thumbnail-or-full-picture" $pictureOptions>: error calling partial: "node_modules\@hyas\images\layouts\partials\helpers\lib-output\image-handling\thumbnail-or-full-picture.html:94:20": execute of template failed: template: partials/hel
pers/lib-output/image-handling/thumbnail-or-full-picture.html:94:20: executing "partials/helpers/lib-output/image-handling/thumbnail-or-full-picture.html" at <partial "helpers/link-open" (dict "page" . "destination" $linkHref "class" $inLinkClass "close" false "target" $inTarget "rel" $rel "destinationNoFind" true)>: error calling partial: partial "helper
s/link-open" not found
Built in 1152 ms
</details>

### Environment

<details>
  <summary>Environment Output</summary>
```
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.

> @hyas/doks@0.5.0 precheck
> npm version

npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
{                          
  '@hyas/doks': '0.5.0',   
  npm: '8.11.0',           
  node: '16.15.1',         
  v8: '9.4.146.24-node.21',
  uv: '1.43.0',            
  zlib: '1.2.11',          
  brotli: '1.0.9',         
  ares: '1.18.1',          
  modules: '93',           
  nghttp2: '1.47.0',       
  napi: '8',               
  llhttp: '6.0.4',         
  openssl: '1.1.1o+quic',
  cldr: '40.0',
  icu: '70.1',
  tz: '2021a3',
  unicode: '14.0',
  ngtcp2: '0.1.0-DEV',
  nghttp3: '0.1.0-DEV'
}

> @hyas/doks@0.5.0 check
> exec-bin node_modules/.bin/hugo/hugo version

hugo v0.101.0-466fa43c16709b4483689930a4f9ac8add5c9f66+extended windows/amd64 BuildDate=2022-06-16T07:09:16Z VendorInfo=gohugoio
</details>

**Comment by danielfdickinson** [comment]
FYI you might want to hold off applying the update for a few hours -- I'm going to be applying the change in more places and may find breakage that my smoke tests missed.

**Comment by h-enk** [comment]
@danielfdickinson, thanks for notifying — I will hold off then.

**Comment by h-enk** [comment]
@danielfdickinson, works with sites using @hyas/images v0.2.2. — see e.g. https://gethyas.com/docs/recipes/images/#image-in-page-bundle. Thanks!

**Comment by danielfdickinson** [comment]
@james-d-elliott @h-enk This is a missing documentation issue on my part.

The default behaviour for the figure shortcode is to ignore static images. To override that you need the `ignoreStaticImages="false"` parameter.

```
{{< figure alt="squarish ASCII art smiley face" src="/substatic/featured-favicon-ascface-512x512.png" ignoreStaticImages="false" >}}
```

There is also a (TODO buried in the source) limitation that the that static directory must be named 'static' and findable find fileExists function. I'm working on removing that limitation by eliminating the attempt to be be smart and detect a missing static image. Instead if `ignoreStaticImages` is `false` then we will accept any `src` and won't try to find the image to verify it exists.

**Comment by danielfdickinson** [comment]
@h-enk 

> However, I noticed that `class` doesn't get applied.

Do you mean that it's not in the HTML or that the CSS is not getting applied (possibly due to elements not being as expected) to the final result?

**Comment by h-enk** [comment]
Ah, sorry, my bad. The class did got added to the `figure` element (but, not to the `img` element as I expected).

But that's maybe even a good a thing, for I could "solve" this for all `figure` elements at once with css (so, without having to add this for every figure element). On the other hand, maybe it would be nice to have the `imageAddClass = "img-fluid lazyload blur-up"` setting in `./config/_default/params.toml` also apply to the figure element?

**Comment by james-d-elliott** [comment]
Also noticed height while it makes it to the element doesn't work, looks like this is due to the doks css however:

```css
figure img {
  margin-bottom: 0.5rem;
  line-height: 1;
  max-width: 100%;
  height: auto;
}
```

**Comment by danielfdickinson** [comment]
Yeah, there are some peculiarities to the way CSS handles height that makes getting it to work as expected is difficult. I haven't figured out a way to set the height but have the width cut-off at page width, without distorting the image. At least with pure CSS.

**Comment by james-d-elliott** [comment]
Your comment actually allowed me to solve my issue, I'm by no means good at CSS/Styles/HTML, but setting the width instead actually fixes my particular issue!
